jamovi/ui/anova.actions.js

```diff
@@ -0,0 +1,87 @@
+
+var view = View.extend({
+
+    initialize: function(ui) {
+        this.calcModelTerms(ui);
+        this.filterModelTerms(ui);
+    },
+
+    events: [
+        {
+            onChange: "ind", execute: function(ui) {
+                this.calcModelTerms(ui);
+            }
+        },
+        {
+            onChange: "modelTerms", execute: function(ui) {
+                this.filterModelTerms(ui);
+            }
+        },
+        {
+            onEvent: "modelTerms.preprocess", execute: function(ui, data) {
+                if (data.intoSelf === false)
+                    data.items = this.getItemCombinations(data.items);
+            }
+        }
+    ],
+
+    calcModelTerms: function(ui) {
+        var variableList = this.cloneArray(ui.ind.value(), []);
+
+        ui.modelSupplier.setValue(this.valuesToItems(variableList, FormatDef.variable));
+
+        var varsDiff = this.findChanges("variableList", variableList, true, FormatDef.variable);
+
+        var termsList = this.cloneArray(ui.modelTerms.value(), []);
+
+        var termsChanged = false;
+
+        for (var i = 0; i < varsDiff.removed.length; i++) {
```

Comment:
can we begin using `let` ... `let`s are super helpful in for-loops like this (reduce errors, etc.)